lesson_7/alex_roman/lesson_7_roman_alex.py

```diff
@@ -0,0 +1,111 @@
+
+# 1. Given the following function:
+
+    #d ef greet(name):
+     # return "Greetings {}!".format(name)
+
+# Create a decorator called uppercase that will uppercase the result
+
+# @uppercase
+# def greet(name):
+    # return "Greetings {}!".format(name)
+
+# print(greet("World"))
+# >>> "GREETINGS WORLD!"
+
+
+def uppercase(func):
+    def inside(name):
+        original=func(name)
+        modified=str.upper(original)
+        return modified
+    return inside
+
+@uppercase
+def greet(name):
+    return "Greetings {}!".format(name)
+
+print(greet("World"))
+
+
+# 2. Given the following function:
+
+    # def divide(first_number, second_number):
+     # return first_number / second_number
+
+# Create a decorator called safe_divide that will output a message if the division cannot be performed, othervise it will return the result.
+
+def safe_divide(func):
+    def inside(numerator, denominator):
+        if denominator==0:
+            print("Can't divide by 0")
+            return
+        func(numerator, denominator)
+    return inside
+
+@safe_divide
+def divide(first_number, second_number):
+    return first_number / second_number
+
+print(divide(10, 0))
```

Comment:
What would be the result if I try the following:
```python
a = divide(10, 2)
print(a * 5)
```
?

Reply:
error ... trying to multiply a NoneType with an integer...
I knew I lost an "Enter" somewhere ... corrected the issue ... now the decorator returns the correct type and value ...